distributed/worker.py

```diff
@@ -1560,12 +1560,7 @@ async def close(  # type: ignore
         # Cancel async instructions
         await BaseWorker.close(self, timeout=timeout)
 
-        teardowns = [
-            plugin.teardown(self)
-            for plugin in self.plugins.values()
-            if hasattr(plugin, "teardown")
-        ]
-        await asyncio.gather(*(td for td in teardowns if isawaitable(td)))
+        await asyncio.gather(*(self.plugin_remove(name) for name in self.plugins))
```

Comment:
Instead of failing on error, this will now just log the exception and try continue with the rest of `close` .

Reply:
I haven't looked deeply but it looks like there are related test failures. For example, `distributed/shuffle/tests/test_shuffle.py::test_closed_worker_during_transfer` (and others) consistently fails with something like:

```
_________________________________________________________________________________________________ test_closed_worker_during_transfer __________________________________________________________________________________________________

c = <Client: No scheduler connected>, s = <Scheduler 'tcp://127.0.0.1:56751', workers: 0, cores: 0, tasks: 0>, a = <Worker 'tcp://127.0.0.1:56752', name: 0, status: closed, stored: 0, running: 0/1, ready: 0, comm: 0, waiting: 0>
b = <Worker 'tcp://127.0.0.1:56755', name: 1, status: closed, stored: 4, running: 1/1, ready: 1, comm: 0, waiting: 0>

    @gen_cluster(client=True, nthreads=[("", 1)] * 2)
    async def test_closed_worker_during_transfer(c, s, a, b):
        df = dask.datasets.timeseries(
            start="2000-01-01",
            end="2000-03-01",
            dtypes={"x": float, "y": float},
            freq="10 s",
        )
        with dask.config.set({"dataframe.shuffle.method": "p2p"}):
            shuffled = df.shuffle("x")
        fut = c.compute([shuffled, df], sync=True)
        await wait_for_tasks_in_state("shuffle-transfer", "memory", 1, b)
        await b.close()

        result, expected = await fut
        dd.assert_eq(result, expected)

        await c.close()
        await check_worker_cleanup(a)
>       await check_worker_cleanup(b, closed=True)

distributed/shuffle/tests/test_shuffle.py:411:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

worker = <Worker 'tcp://127.0.0.1:56755', name: 1, status: closed, stored: 4, running: 1/1, ready: 1, comm: 0, waiting: 0>, closed = True, interval = 0.01, timeout = 5

    async def check_worker_cleanup(
        worker: Worker,
        closed: bool = False,
        interval: float = 0.01,
        timeout: int | None = 5,
    ) -> None:
        """Assert that the worker has no shuffle state"""
        deadline = Deadline.after(timeout)
>       plugin = worker.plugins["shuffle"]
E       KeyError: 'shuffle'

distributed/shuffle/tests/test_shuffle.py:98: KeyError
```

It's just a guess, but it looks like our `check_worker_cleanup` utility is expecting there to be a `shuffle` plugin that we're now removing with the change here (again, haven't looked that closely)

Reply:
Yeah, that was an unexpected side effect, that's fixed now.